presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/utils/delegates/EcoCompositeDisposableProperty.kt

```diff
@@ -0,0 +1,22 @@
+package com.exallium.mvvmcleanexample.presentation.utils.delegates
+
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * A property delegate for [Disposable] objects that performs cleanup on assignment.
+ *
+ * When a new value is assigned, [setValue] will call [Disposable.dispose] on the previous value if available.
+ * @param disposable the new value
+ */
+class EcoCompositeDisposableProperty(private var disposable: CompositeDisposable) : ReadWriteProperty<Any?, CompositeDisposable> {
```

Comment:
Maybe not the best name, couldn't think of anything else short that made sense.
These 'eco' property delegates are intended to perform some kind of cleanup of their previous value on reassignment. 

Though I think the 'eco' prefix can be dropped come to think of it.